coalib/bears/requirements/PythonRequirement.py

```diff
@@ -0,0 +1,81 @@
+
+
+class PythonRequirement(PackageRequirement):
+    """
+    This class is a subclass of ``PackageRequirement``, and helps specifying
+    requirements from ``pip``, without using the manager name.
+    """
+
+    def __init__(self, package, version=""):
+        """
+        Constructs a new PythonRequirement, using the PackageRequirement
+        constructor.
+
+        :param package: A string with the name of the package to be installed.
+        :param version: A number that contains the version. Leave empty to
+                        install latest.
+        """
+        PackageRequirement.__init__(self, 'pip', package, version)
+
+    @classmethod
+    def multiple(cls, *args):
+        """
+        Creates a tuple of multiple ``PythonRequirements``.
+
+        You should use the ``multiple`` method if you have more
+        requirements from the same manager. This can receive both tuples of
+        strings, in case you want a specific version, or a simple string, in
+        case you want the latest version to be installed.
+
+        This is the case where you would provide strings only, to install the
+        latest version automatically:
+
+        >>> REQUIREMENTS = PythonRequirement.multiple(
+        ...             'coala_decorators', 'setuptools')
+
+        And if you choose to mix them, specifying version for some and for some
+        not:
+
+        >>> REQUIREMENTS = PythonRequirement.multiple(
+        ...             'coala_decorators', ('setuptools', '19.2'))
+
+        In case you provide too many arguments into the tuple, an error will be
+        raised:
+
+        >>> REQUIREMENTS = PythonRequirement.multiple(
+        ...             'coala_decorators', ('setuptools', '19.2', 'colorama'))
+        Traceback (most recent call last):
+        ...
+        TypeError: The tuple must have 2 elements.
+
+        The same would happen in case you provide something different than a
+        string or a tuple:
+
+        >>> x = [1, 2, 3, 4]
+        >>> REQUIREMENTS = PythonRequirement.multiple(x)
+        Traceback (most recent call last):
+        ...
+        TypeError: The arguments need to be tuples or strings.
+
+        :param args:       Should be tuples of strings: ``('packageName',
+                           'version')`` or strings: ``'packageName'`` if latest
+                           version is wanted.
+        :return:           A tuple containing ``PythonRequirements``.
+        :raises TypeError: In case the tuples contain with more or less than
+                           two elements. Also raised when arguments are neither
+                           tuples nor strings.
+        """
+        reqs = ()
+        for requirement in args:
+            if isinstance(requirement, str):
+                reqs += (cls(requirement),)
+            elif isinstance(requirement, tuple):
+                try:
+                    name, version = requirement
+                    reqs += (cls(name, version),)
```

Comment:
oh shit, using a list and then collecting is way faster:
```python
>>> from timeit import *
>>> def list_then_collect():
...  x = []
...  for i in range(1000):
...   x.append(i)
...  return tuple(x)
...
>>> def direct_tuple():
...  x = ()
...  for i in range(1000):
...   x += (i,)
...  return x
...
>>> dt = Timer(direct_tuple)
>>> ltc = Timer(list_then_collect)
>>> dt.timeit(1000)
1.2906605905547812
>>> ltc.timeit(1000)
0.10712832800553862
>>> ltc.timeit(1000)
0.10652603528101068
>>> dt.timeit(1000)
1.1920436711211142
```
12x slower with using tuples, so we have learnt as a design consideration: appending to tuples sucks! Better using a list and then collecting!